hack/make-rules/clean.sh

```diff
@@ -24,7 +24,7 @@ source "${KUBE_ROOT}/hack/lib/util.sh"
 CLEAN_PATTERNS=(
   "_tmp"
   "doc_tmp"
-  ".*/zz_generated.openapi.go"
+  "((?!staging\/src\/k8s\.io\/apiextensions-apiserver\/pkg\/generated\/openapi).)*/zz_generated.openapi.go"
```

Comment:
A nit which doesn't really matter since we only have one of these in staging: this path should probably be general enough to find everything the diff is going to consider. (same in the .gitignore file)